File: window.nim
import sdl2

discard init(INIT_EVERYTHING)

let win = createWindow(
  "Window example",
  SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
  400, 400,
  SDL_WINDOW_SHOWN
)

var
  ev: Event
  running = true

while running:
  while ev.pollEvent():
    if ev.kind == QuitEvent: running = false

win.destroy()
sdl2.quit()

File: window.c
#include <SDL.h>
#include <stdbool.h>

#undef main

int main(void) {
  SDL_Init(SDL_INIT_EVERYTHING);

  SDL_Window* win;
  SDL_Event ev;
  bool running = true;

  win = SDL_CreateWindow(
    "Window example",
    SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
    400, 400,
    SDL_WINDOW_SHOWN
  );

  while (running) {
    while (SDL_PollEvent(&ev) > 0) {
      if (ev.type == SDL_QUIT) running = false;
    }
  }

  SDL_DestroyWindow(win);
  SDL_Quit();

  return 0;
}